### PR TITLE
test(Slider): add test cases, close#346

### PR DESCRIPTION
**Pull request by youluna**

### CODE REVIEW

File: src/slider/slick/dots.jsx [hunk 1]
```diff
@@ -66,18 +81,25 @@ export default class Dots extends React.Component {
...
-                        (dotsRender instanceof Function) ?
-                            <span>
-                                {dotsRender(docIndex, currentSlideIndex)}
-                            </span> :
-                            <button />
-                    }
+                <li
+                    key={i}
+                    className={itemClazz}
+                    role="listitem"
```
Comment (jdkahn) [inline src/slider/slick/dots.jsx]: `role=listitem` is implicit for `<li>` with a `<ul>` or `<ol>` parent. Its not recommended to have it.

Comment (jdkahn) [inline src/slider/slick/dots.jsx]: `role=listitem` is implicit for `<li>` with a `<ul>` or `<ol>` parent. Its not recommended to have it.

File: src/slider/slick/dots.jsx [hunk 2]
```diff
@@ -66,18 +81,25 @@ export default class Dots extends React.Component {
...
+                    {...handleProp}
+                >
+                    {dotsRender instanceof Function ? (
+                        <span>{dotsRender(docIndex, currentSlideIndex)}</span>
+                    ) : (
+                        <button tabIndex="-1" />
```
Comment (jdkahn) [inline src/slider/slick/dots.jsx]: When using a discouraged practice, like setting `tabIndex="-1"` its helpful to have a comment why. Will help the next maintainer. 
Suggested comment: `Slider is navigated by right and left arrow buttons so the dots are not required functionality`

File: src/slider/slick/track.jsx [hunk 3]
```diff
@@ -161,17 +199,20 @@ const renderSlides = (specProps) => {
...
-        prefix: 'next-'
+        prefix: 'next-',
     };
 
-    render () {
+    render() {
         const slides = renderSlides(this.props);
         return (
-            <div className={`${this.props.prefix}slick-track`} style={this.props.trackStyle}>
+            <div
```
Comment (jdkahn) [inline src/slider/slick/track.jsx]: if the slides have `role="listitem"` then this should be `ol`, `ul`, or `role="list"`

File: src/slider/slider.jsx [hunk 4]
```diff
@@ -231,11 +244,22 @@ export default class Slider extends Component {
 
         return (
             <ConfigProvider rtl={false}>
-                <div role="toolbar" dir="ltr" className={clazz} style={style} {...obj.pickOthers({...Slider.propTypes, ...InnerSlider.propTypes}, sliderProps)}>
-                    <InnerSlider ref={InnerSlider => (this.innerSlider = InnerSlider)} {...sliderProps} />
+                <div
+                    role="list"
+                    dir="ltr"
```
Comment (jdkahn) [inline src/slider/slider.jsx]: not related to this change, but why is the `dir` fixed to `ltr`?

File: test/slider/a11y-spec.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        if (wrapper) {
+            wrapper.unmount();
+            wrapper = null;
+        }
+        unmount();
+    });
+
+    it('should not have any violations', async () => {
+        wrapper = await testReact(
+            <div>
```
Comment (jdkahn) [inline test/slider/a11y-spec.js]: is the div parent necessary?

File: test/slider/a11y-spec.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                <Slider>{slides}</Slider>
+            </div>
+        );
+        return wrapper;
+    });
+
+    it('should render single slider', async () => {
```
Comment (jdkahn) [inline test/slider/a11y-spec.js]: We aren't testing render, but maybe the `should not have any violations` prefix (while accurate) is too long for each test. If you can think of a shorter one use that, if not change to `should not have any violations for single slider`. And same for following tests

File: test/slider/index-spec.js [hunk 7]
```diff
@@ -45,6 +45,16 @@ describe('slider', function() {
             assert(wrapper.find('.next-slick-dots').length === 0);
         });
 
+        it('should dots render', () => {
```
Comment (jdkahn) [inline test/slider/index-spec.js]: `should render dots`

File: src/slider/slick/dots.jsx (hunk 1 above)
Comment (youluna) [reply src/slider/slick/dots.jsx]: done

File: src/slider/slick/dots.jsx (hunk 2 above)
Comment (youluna) [reply src/slider/slick/dots.jsx]: OK~

File: src/slider/slider.jsx (hunk 4 above)
Comment (youluna) [reply src/slider/slider.jsx]: There are a lot of calculations in Slider, `dir` fixed to `ltr` to simplify the calculations when rtl.

File: test/slider/a11y-spec.js (hunk 5 above)
Comment (youluna) [reply test/slider/a11y-spec.js]: I removed it

**Comment by youluna** [comment]
@jdkahn I've made all the changes, please review~

**Pull request merged**